04-Lab/FunctionalSpatialImageProcessing.ipynb

```diff
@@ -0,0 +1,517 @@
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "# change the image type to Mask/Label in fslview (remember where to look?)\n",
+    "# also change the colorscale from grayscale to render or random-rainbow\n",
+    "fslview HarvardOxford-sub-maxprob-thr50-2mm.nii.gz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Example for extracting the left putamen (intensity 6)\n",
+    "# review: what does -bin do?\n",
+    "fslmaths HarvardOxford-sub-maxprob-thr50-2mm.nii.gz -thr 6 -uthr 6 -bin Lputamen.nii.gz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Did it work?\n",
+    "fslview MNI152_T1_2mm_brain.nii.gz Lputamen.nii.gz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Finally we can use apply warp to apply both our nonlinear and affine warps at once to our functional\n",
+    "# This is typically done when we want to put a regional mask in standard space into functional space \n",
+    "# e.g., for our atlas-based ROI of the caudate\n",
+    "# interp=nn means the interpolation method we used is nearest neighbor.\n",
+    "# nearest neighbor: this means a voxel takes the value of it's closest neighbor\n",
+    "applywarp -i Lputamen -r example_func_brain -o Lputamen_mask_func -w MNI_2_T1_warp --postmat=T12func_6dof.mat --interp=nn"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Did it work? \n",
+    "fslview example_func.nii.gz Lputamen_mask_func.nii.gz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot a more static image with some transparency using alpha setting\n",
+    "fimg=\"example_func.nii.gz\"\n",
+    "roi=\"Lputamen_mask_func.nii.gz\"\n",
+    "plotting.plot_roi(roi,fimg, display_mode=\"z\", cut_coords=10, alpha=.5, title=\"Lputamen to func\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Why can't we use this to extract the timeseries from this region in the my_bold.nii.gz image yet?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problem sets"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*1a)* Use the epi_reg tool to generate a transform from functional to T1 space, and combine these transforms to complete a registration of the functional to MNI using BBR for the functional step.\n",
```

Comment:
awesome questions, I think they should be able to handle using a command we haven't explicitly done in lab now.